packages/plot/src/plots.rs

```diff
@@ -0,0 +1,194 @@
+use plotters::prelude::*;
+use std::ops::Range;
+
+use crate::eval::*;
+use crate::utils::*;
+
+type Points = Vec<(f64, f64)>;
+type Plot = Vec<Points>;
+
+fn get_function_plots(ctx: &mut PlotContext) -> Result<Vec<Plot>, String> {
+    let mut plots = Vec::new();
+
+    for function in ctx.data.split('\n') {
+        let mut x_iter = ctx.rm.get_point_range('x')?;
```

Comment:
I encountered an issue were a trailing newline or whitespace character would cause the module to fail. Maybe add something along these lines. Note: I've not tested my suggestion, so it might not compile. But you get the idea :).
```suggestion
        // skip lines only containing whitespace
        if function.chars().all(char::is_whitespace) {
            continue;
        }

        let mut x_iter = ctx.rm.get_point_range('x')?;
```